### PR TITLE
Add convective gust contribution to U10

### PR DESCRIPTION
**Pull request by megandevlan**
### Description of changes
Modifications made to enhance U10 computed in shr_flux_mod based on convective rain rate from CAM. Additional modifications made to output the magnitude of that enhancement (UGUST) in CAM as well. The option to turn on the enhancement is controlled by a name list variable, add_gusts. 

### Specific notes

Contributors other than yourself, if any: @swrneale

CMEPS Issues Fixed (include github issue #): https://github.com/ESCOMP/CMEPS/issues/267

Are changes expected to change answers? (specify if bfb, different at roundoff, more substantial) None if add_gusts=False

Any User Interface Changes (namelist or namelist defaults changes)? add_gusts option in namelist; default is false. 

### Testing performed
Please describe the tests along with the target model and machine(s) 
If possible, please also added hashes that were used in the testing

Tests on Derecho as part of most recent cam-dev: https://github.com/NCAR/amwg_dev/issues/440

### CODE REVIEW

File: cesm/flux_atmocn/shr_flux_mod.F90 [hunk 1]
```diff
@@ -257,22 +263,21 @@ SUBROUTINE flux_atmOcn(logunit, nMax  ,zbot  ,ubot  ,vbot  ,thbot ,   &
     real(R8)    :: tdiff(nMax)               ! tbot - ts
     real(R8)    :: vscl
 
+    real(R8)    :: ugust      ! function: gustiness as a function of convective rainfall.  
+    real(R8)    :: gprec   ! dummy arg ~ ?
```
Comment (jedwards4b) [inline cesm/flux_atmocn/shr_flux_mod.F90:267]: Not a great comment here.

File: cesm/flux_atmocn/shr_flux_mod.F90 [hunk 2]
```diff
@@ -257,22 +263,21 @@ SUBROUTINE flux_atmOcn(logunit, nMax  ,zbot  ,ubot  ,vbot  ,thbot ,   &
...
-    cdn(Umps)  =  0.0027_R8 / min(33.0000_R8,Umps) + 0.000142_R8 + &
-         0.0000764_R8 * min(33.0000_R8,Umps) - 3.14807e-13_r8 * min(33.0000_R8,Umps)**6
-    ! Capped Large and Pond by wind
-    !   cdn(Umps)  =   0.0027_R8 / min(30.0_R8,Umps) + 0.000142_R8 + 0.0000764_R8 * min(30.0_R8,Umps) 
-    ! Capped Large and Pond by Cd
-    !   cdn(Umps) = min(0.0025_R8, (0.0027_R8 / Umps + 0.000142_R8 + 0.0000764_R8 * Umps ))
-    ! Large and Pond
-    !   cdn(Umps)  =   0.0027_R8 / Umps + 0.000142_R8 + 0.0000764_R8 * Umps
-
+    cdn(Umps)  =   0.0027_R8 / Umps + 0.000142_R8 + 0.0000764_R8 * Umps
```
Comment (jedwards4b) [inline cesm/flux_atmocn/shr_flux_mod.F90:270]: You removed the cap on this equation so this will affect not just cam_dev but all configurations?

**Review by jedwards4b — CHANGES_REQUESTED** [review]
The cdn variable will change more than just cam_dev - can you clarify this is what you want.   Currently the add_gust variable is set to false and the only way to change it is in user_nl_cam or in a testmod.   I think you can set an esmf info object in cam to the result of cam_physpkg_is("cam_dev") then query that info object in cmeps to determine if the add_gust variable should be used.

File: cesm/flux_atmocn/shr_flux_mod.F90 (hunk 2 above)
Comment (megandevlan) [reply cesm/flux_atmocn/shr_flux_mod.F90]: Thanks Jim, fixed this now; not something I intended to change.

File: cesm/flux_atmocn/shr_flux_mod.F90 (hunk 1 above)
Comment (megandevlan) [reply cesm/flux_atmocn/shr_flux_mod.F90]: Updated this too; working on the esmf info object now.

**Comment by megandevlan** [comment]
I'd been setting the namelist add_gust in usr_nl_cpl instead of usr_nl_cam; is the latter better practice?

**Comment by jedwards4b** [comment]
No, I think that user_nl_cpl is the correct place, I was mistaken in my previous comment.

**Comment by megandevlan** [comment]
Thanks Jim. I'm somewhat stumped for how to set an esmf object in CAM that can then be queried in ESMF; do you have an example of something similar being done that I could follow for some other variable? I'm guessing it's something that would be in the atm_comp_nuopc file in CAM?

**Comment by jedwards4b** [comment]
@megandevlan I'm looking for the best solution and will get back to you.

**Comment by jedwards4b** [comment]
@megandevlan I've opened [a PR](https://github.com/ESCOMP/CMEPS/pull/423) to your fork that addresses the cam-dev issue, since cam_dev is a compile time option and not a run time option we can do this check in the cmeps buildnml.   The PR also fixes the .github testing, so once it's merged all tests should pass.

**Comment by megandevlan** [comment]
Thanks @jedwards4b, especially for getting to this so quickly!! These fixes looks great to me.

**Review by jedwards4b — APPROVED** [review]

**Comment by cacraigucar** [comment]
@jedwards4b - Is there a tag for this change?

**Comment by jedwards4b** [comment]
cmeps0.14.48  Tags are automatically generated in this repo, do a git fetch to see new tags.

**Comment by jedwards4b** [comment]
cmeps0.14.49 now.

**Pull request merged**